### DISCUSSION TITLE
Lesson 6: Error Compiling MockV3Aggregator

### DISCUSSION THREAD

**Discussion by karlMoreno** [smartcontractkit/full-blockchain-solidity-course-py#831] [Q&A] [answered]
Hi! I went to compile MockV3Aggregator but was hit with a solidity version error saying I was compiling from 8.11 when my settings and my remote was compiled to 0.6.0. Curiously enough, this was fixed when in my settings.json in my .vscode folder and my config file when I added this:

dependencies:
  # - <organization/repo>@<version>
  - smartcontractkit/chainlink-brownie-contracts@1.1.1
  - smartcontractkit/chainlink-mix@2.3.0
compiler:
  solc:
    remappings:
      - '@chainlink=smartcontractkit/chainlink-brownie-contracts@1.1.1'
      - '@chainlink=smartcontractkit/chainlink-mix@2.3.0'
   

Now all I get is the error:
CompilerError: solc returned the following errors:

contracts/test/MockV3Aggregator.sol:4:1: ParserError: Source "/root/.brownie/packages/smartcontractkit/chainlink-mix@2.3.0/contracts/src/v0.6/interfaces/AggregatorV2V3Interface.sol" not found: File not found.
import "@chainlink/contracts/src/v0.6/interfaces/AggregatorV2V3Interface.sol";
^----------------------------------------------------------------------------^

**Comment by cromewar** [comment]
Hello @karlMoreno If you take a look at chainlink mix: https://github.com/smartcontractkit/chainlink-mix you can see that direction does not exists, could you please share your brownie config.yaml?

**Reply by karlMoreno** [reply]
This is the brownie-config.yaml file


<img width="1357" alt="Screen Shot 2022-01-21 at 11 28 49 AM" src="https://user-images.githubusercontent.com/86685561/150588515-5b98f80d-755f-4b13-9db2-98a09c582681.png">


When I remove everything else and write brownie compile in my terminal I get:

ParserError: Source file requires different compiler version (current compiler is 0.8.11+commit.d7f03943.Linux.g++) - note that nightly builds are considered to be strictly less than the released version
 --> /root/.brownie/packages/smartcontractkit/chainlink-brownie-contracts@0.3.1/contracts/src/v0.6/vendor/SafeMathChainlink.sol:2:1:
  |
2 | pragma solidity ^0.6.0;
  | ^^^^^^^^^^^^^^^^^^^^^^^

despite being in the 0.6.0 remote global compiler version




so when I right click and hit compile with configured remote version I get this:

Retrieving compiler information:
Compiler using remote version: 'v0.6.0+commit.26b70077', solidity version: 0.6.0+commit.26b70077.Emscripten.clang
/Users/karlmoreno/work/Blockchain/demos/brownie_fund_me/contracts/test/MockV3Aggregator.sol:4:1: ParserError: Source "@chainlink/contracts/src/v0.6/interfaces/AggregatorV2V3Interface.sol" not found: File import callback not supported
import "@chainlink/contracts/src/v0.6/interfaces/AggregatorV2V3Interface.sol";
^----------------------------------------------------------------------------^

Compilation failed with 1 errors

**Reply by karlMoreno** [reply]
I fixed the compiler version by setting sold to 0.6.0 in my brownie-config.yaml file:
<img width="1318" alt="Screen Shot 2022-01-21 at 2 52 54 PM" src="https://user-images.githubusercontent.com/86685561/150610526-1fa27cc8-bd83-4230-9197-fffc78946173.png">

but now i get this error:
<img width="1297" alt="Screen Shot 2022-01-21 at 2 53 33 PM" src="https://user-images.githubusercontent.com/86685561/150610559-fe82954c-4dd4-4331-a997-36504b625291.png">

and it doest work with chainlink-brownie-contract versions 0.2.2 or 0.3.1

**Comment by karlMoreno** [comment, marked as answer]
Issue resolved when in brownie-config.yaml file i set

solc:
    version 0.6.6

and in the the MockV3Aggregator file I changed from:

pragma solidity ^0.6.0;

to:

pragma solidity >=0.6.0;

**Reply by cromewar** [reply]
Nice catch; I'm glad is solved now.
